File: Top_Interview_150/Hashmap/python/219.py
from typing import List


class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        dct = dict()
        for i in range(len(nums)):
            if nums[i] in dct:
                if i - dct[nums[i]] <= k:
                    return True
            dct[nums[i]] = i

        return False

File: Top_Interview_150/Hashmap/cpp/219.cpp
#include <vector>
#include <unordered_map>

using namespace std;

class Solution
{
public:
    bool containsNearbyDuplicate(vector<int> &nums, int k)
    {
        unordered_map<int, int> dct;
        for (int i = 0; i < nums.size(); ++i)
        {
            if (dct.find(nums[i]) != dct.end())
                if (i - dct[nums[i]] <= k)
                    return true;
            dct[nums[i]] = i;
        }

        return false;
    }
};